ext/checkers.jl

```diff
@@ -0,0 +1,230 @@
+        !same_short_name &&
+        @info "Short names are not the same between the OutputVar ($var_short_name) and metadata ($metadata_short_name)"
+    return same_short_name
+end
+
+"""
+    Checker.check(
+        ::DimNameChecker,
+        var::OutputVar,
+        metadata::Metadata;
+        verbose = false,
+    )
+
+Return `true` if `var` and `metadata` have the same dimensions, `false`
+otherwise.
+"""
+function Checker.check(
+    ::DimNameChecker,
+    var::OutputVar,
+    metadata::Metadata;
+    verbose = false,
+)
+    var_dim_names = ClimaAnalysis.conventional_dim_name.(keys(var.dims))
+    metadata_dim_names =
+        ClimaAnalysis.conventional_dim_name.(keys(metadata.dims))
+    same_dim_names = issetequal(var_dim_names, metadata_dim_names)
+    verbose &&
+        !same_dim_names &&
+        @info "Dimensions are not the same between the OutputVar ($var_dim_names) and metadata ($metadata_dim_names)"
+    return same_dim_names
+end
+
+"""
+    Checker.check(
+        ::DimUnitsChecker,
+        var::OutputVar,
+        metadata::Metadata;
+        verbose = false,
+    )
+
+Return `true` if the units of the dimensions in `var` and `metadata` are the
+same, `false` otherwise. This function assumes `var` and `metadata` have the
+same dimensions.
+"""
+function Checker.check(
+    ::DimUnitsChecker,
+    var::OutputVar,
+    metadata::Metadata;
+    verbose = false,
+)
+    for var_dim_name in keys(var.dims)
+        md_dim_name = ClimaAnalysis.Var.find_corresponding_dim_name_in_var(
+            var_dim_name,
+            metadata,
+        )
+        var_dim_units = ClimaAnalysis.dim_units(var, var_dim_name)
+        md_dim_units = ClimaAnalysis.dim_units(metadata, md_dim_name)
+        same_dim_units = var_dim_units == md_dim_units
+        verbose &&
+            !same_dim_units &&
+            @info "Units of dimension ($var_dim_name) are not the same between OutputVar ($var_dim_units) and metadata ($md_dim_units)"
+        same_dim_units || return false
+        if var_dim_units == "" || md_dim_units == ""
+            @warn(
+                "Units for $(ClimaAnalysis.conventional_dim_name(var_dim_name)) is missing in var or metadata"
+            )
+        end
+    end
+    return true
+end
+
+"""
+    Checker.check(
+        ::UnitsChecker,
+        var::OutputVar,
+        metadata::Metadata;
+        verbose = false,
+    )
+
+Return `true` if `var` and `metadata` have the same units, `false` otherwise.
+"""
+function Checker.check(
+    ::UnitsChecker,
+    var::OutputVar,
+    metadata::Metadata;
+    verbose = false,
+)
+    var_units = ClimaAnalysis.units(var)
+    metadata_units = ClimaAnalysis.units(metadata)
+    if var_units == "" || metadata_units == ""
+        @warn("Var or metadata may be missing units")
+    end
+    same_units = var_units == metadata_units
+    verbose &&
+        !same_units &&
+        @info "Units of OutputVar ($var_dim_name) are not the same as units of metadata ($var_dim_units)"
```

Comment:
```suggestion
@info "Units of OutputVar ($var_units) are not the same as units of metadata ($metadata_units)"
```

I think this is a typo